### PR TITLE
[s]Shadowling re-balance and endgame rework.

### PR DESCRIPTION
**Pull request by kilkun**
Hey lads, it's ya boy Kilkun back on another PR. This time I got something huge for you lads.
- Changes multiple low-level shadowling spells to have shorter recharges (namely blinding smoke)
- Made flares and their children susceptible to their magic "who turned out the lights" spell
- Massively buffed the Shadowling's get-away jaunt by changing it from 4 seconds to 8.
- Wizards can no longer be annihilated.
- Added new announcement sound.
- 50% chance of an ascendant hunter wizard spawning on ascendance.
- New special veil render available only to these wizard hunters. Actually gives the veil render a purpose.

Hopefully this makes the gamemode fun, by making it completely jump the shark. Keep that in mind before you get all salty in the discussion, that excitement and a real end game is lacking from shadowling.

Also this completely canon image is the preamble to the endgame of shadowling. 100% free of charge.

This also makes the thralls more relevant in the end game, as the ascendant is nearly powerless against the wizard now, having to rely on it's thralls to protect it until it can escape to centcom.

### CODE REVIEW

**Comment by TehZombehz** [comment]
I just kinda wish shadowlings would have disappeared forever.

**Comment by TechnoAlchemisto** [comment]
this is an okay PR which could be made better by commenting shadowlings out of the code completely

**Comment by kilkun** [comment]
it's okay because this pr is actually a secret attempt to just get more wizard rounds, only they are disguised as shadowling rounds.

I'm debating on making the veil render's result player controlled, too.

File: code/game/objects/items/dehy_carp.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@
 /obj/item/toy/carpplushie/dehy_carp
 	var/mob/owner = null	//Carp doesn't attack owner, set when using in hand
 	var/owned = 0	//Boolean, no owner to begin with
+	var/mobtype = /mob/living/simple_animal/hostile/carp //So admins can change what mob spawns via var fuckery
```
Comment (KorPhaeron) [inline code/game/objects/items/dehy_carp.dm]: ?

Comment (kilkun) [inline code/game/objects/items/dehy_carp.dm]: wait what? that pr was already merged wasn't it?

File: code/game/gamemodes/wizard/artefact.dm [hunk 2]
```diff
@@ -82,6 +82,55 @@
...
+
+/obj/item/weapon/veilrender/uncharged/attack(mob/living/M, mob/living/user)
+	if(!iswizard(user))
+		user.Paralyse(5)
+		user.adjustBrainLoss(5)//Aay it's an eldritch artifact, I ain't gotta explain shit!
+		user << "<span class='userdanger'>Your mind oozes as you try to figure out how to use the sanity-shattering blade!</span>"
+		return ..()
+	if(!istype(M, /mob/living/simple_animal/ascendant_shadowling))
+		return ..()
+	if(captured_ascendants<=2)
```
Comment (Alek2ander) [inline code/game/gamemodes/wizard/artefact.dm]: Looks like the wrong comparision. >= 2?

File: code/game/gamemodes/wizard/artefact.dm [hunk 3]
```diff
@@ -82,6 +82,55 @@
...
+	rend_desc = "You hear the sound of laughing slaughter demons."
+	spawn_fast = 1
+	var/captured_ascendants = 0
+
+/obj/item/weapon/veilrender/uncharged/attack_self(mob/user)
+	if(charges > 0 && captured_ascendants==2)
+		new /obj/effect/rend(get_turf(user), spawn_type, spawn_amt, rend_desc, spawn_fast)
+		charges--
+		captured_ascendants = 0
+		user << "<span class='boldannounce'>You combine the awesome power of the three ascendants bend the bridge of reality, beckoning forth the blood geometer.</span>"
```
Comment (Alek2ander) [inline code/game/gamemodes/wizard/artefact.dm]: >captured_ascendants==2

That would be two ascendants, not three

**Comment by KorPhaeron** [comment]
I think the wizard is a fun idea but I don't think normal shadowlings need to be stronger

**Comment by Zilenan91** [comment]
If it makes wizards occur more often it's a noble goal :+1: 

Also top quality MS Paint art :+1:  for that alone

**Comment by Fox-McCloud** [comment]
Why buff shadowlings. Isn't there win rate already patently ridiculous?

**Comment by PKPenguin321** [comment]
Dat art
Dem buffs
Deeze wizards
:+1: This is excellente

**Comment by Durkel** [comment]
Nah dawg, nah. No shadowling buffs, they're already ridiculous with their reee,shit smoke, and jaunt combo making them a bitch to chase.

**Comment by Dorsisdwarf** [comment]
:-1: making flares vulnerable to LIGHTS OUT  YOURE AN ANTAG NOW

**Comment by kilkun** [comment]
Alright, I'll comprimise: I toned down quite a few of the buffs. Kept most of the range buffs and still reduced the recharge timers on the less griffy spells, but I think you'll find this is far more agreeable.

Also I fixed some logic errors and made it impossible to hypno/shock wizards too.

File: code/game/gamemodes/wizard/artefact.dm (hunk 3 above)
Comment (kilkun) [inline code/game/gamemodes/wizard/artefact.dm]: I know. If the wizard succeeds and eats the two ascendants, I want to evoke a sense of terror in the last one with the arrival of a real god. The kind of terror you can only achieve by stripping away the comfortable safety that a player is given when they play an overpowered role.

File: code/game/gamemodes/shadowling/special_shadowling_abilities.dm [hunk 4]
```diff
@@ -155,6 +155,48 @@ var/list/possibleShadowlingNames = list("U'ruan", "Y`shej", "Nex", "Hel-uae", "N
...
+								spawn(0)
+									switch(alert(G, "Do you wish to be considered for the special position of the Space Wizard researcher?","Please answer in 10 seconds!","Yes","No"))
+										if("Yes")
+											if((world.time-time_passed)>100)//If more than 10 game seconds passed.
+												return
+											candidates += G
+										if("No")
+											return
+										else
+											return
```
Comment (Dorsisdwarf) [inline code/game/gamemodes/shadowling/special_shadowling_abilities.dm]: Isn't there a proc for this now?

File: code/game/gamemodes/wizard/artefact.dm [hunk 5]
```diff
@@ -82,6 +82,55 @@
...
+		user.visible_message("<span class='boldannounce'>[src] hums with power as [user] deals a blow to [activate_descriptor] itself!</span>")
+		world << "span class='shadowling'>NAR`SIE  WGHA'N SHUGG-YAR!</span>"
```
Comment (Dorsisdwarf) [inline code/game/gamemodes/wizard/artefact.dm]: Span_class narsie plz

**Comment by Xhuis** [comment]
If I still cared about shadowlings, I would be angry at this PR and demand that you close it. However, I'm aiming for #16441 to replace shadowling as a gamemode, so this is fine I suppose.

File: code/game/gamemodes/shadowling/special_shadowling_abilities.dm [hunk 6]
```diff
@@ -155,6 +155,48 @@ var/list/possibleShadowlingNames = list("U'ruan", "Y`shej", "Nex", "Hel-uae", "N
 					M << "<span class='userdanger'>An immense pressure slams you onto the ground!</span>"
 				world << "<font size=5><span class='shadowling'><b>\"VYSHA NERADA YEKHEZET U'RUU!!\"</font></span>"
 				world << 'sound/hallucinations/veryfar_noise.ogg'
+				if(!ticker.mode.wizards)
+					message_admins("A space wizard has spawned to try and combat the Ascendant.")
+					priority_announce("ERROR \[0x0441c\]: MEMORY HIJACK EXECUTED. UNAUTHORIZED ANNOUNCEMENT DETECTED.", null, 'sound/misc/wizard_announce.ogg')
```
Comment (coiax) [inline code/game/gamemodes/shadowling/special_shadowling_abilities.dm]: So wait, is there an actual announcement made to the crew?

File: code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 7]
```diff
@@ -61,7 +61,7 @@
 	clothes_req = 0
 	range = 5
 	action_icon_state = "veil"
-	var/blacklisted_lights = list(/obj/item/device/flashlight/flare, /obj/item/device/flashlight/slime)
+	var/blacklisted_lights = list(/obj/item/device/flashlight/slime)//Slightly buffed, requiring slimes instead of slimes and flares
```
Comment (coiax) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: > slightly buffed

I can tell you I've never seen any defence against shadowlings that didn't involve botany spamming torches. Taking out botany is one of the key points to a shadowling victory.

I'm all for Xeno slimelights getting more love, but DAMN son. At least give people the ability to build a bonfire out of broken wooden chairs to huddle around.

**Comment by Supermichael777** [comment]
> unironically buffing shadowing, a literally cancerous game-mode.
top kek.
Flares being immune is intentional.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 7 above)
Comment (Supermichael777) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: the thing about shadowlings is that they cant put out fire or bio-luminescence for deep lore reasons.

File: code/game/gamemodes/shadowling/special_shadowling_abilities.dm (hunk 4 above)
Comment (kilkun) [inline code/game/gamemodes/shadowling/special_shadowling_abilities.dm]: Not that I know of, I just replicated the one-click wizard code.

File: code/game/gamemodes/wizard/artefact.dm (hunk 5 above)
Comment (kilkun) [inline code/game/gamemodes/wizard/artefact.dm]: It's one of the shadowlings screaming before it gets eaten.

File: code/game/gamemodes/shadowling/special_shadowling_abilities.dm (hunk 6 above)
Comment (kilkun) [inline code/game/gamemodes/shadowling/special_shadowling_abilities.dm]: Yes, complete with a new announcement sound. This gives the ascendants time for their thralls to prepare to defend it.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 7 above)
Comment (kilkun) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: Keep in mind that there is now a second part to each shadowling round, the buffs were created with this in mind.

Ideally as soon as shadowlings are screamed over the radio (and assuming botany was assimilate), assistants/the warden should already be planting glowshrooms everywhere.

**Comment by Archie700** [comment]
>buffing shadowlings just to get more wizard rounds
>buffing shadowlings without nerfing other features
>buffing shadowlings period

There are many whys to this PRs that I can say.

**Comment by coiax** [comment]
When a shadowling ascends though, that's a working time for the hunterwizard of about 3 minutes, and whatever time it takes to speed launch the shuttle, assuming that the ascended shadowlings can just thrall anyone in power. That seems like a really small time for the wizards to find and capture the ascended, even with a whole host of wizard powers.

Also, can't shadowlings hide by going permamently phase shifted?

File: code/game/gamemodes/shadowling/special_shadowling_abilities.dm (hunk 4 above)
Comment (JJRcop) [inline code/game/gamemodes/shadowling/special_shadowling_abilities.dm]: ya uh https://github.com/tgstation/-tg-station/blob/60237bbb31b309480d5a1cf1890a3afec5c58dec/code/__HELPERS/game.dm#L394

**Comment by kilkun** [comment]
Last time I played shadowling, phase shift was broken. And if that's not the case the wizard can always just murder all of his thralls.

**Comment by Archie700** [comment]
Or shock wizards. Or enthrall them from far. Or explode them. Or any combination.

**Comment by kilkun** [comment]
None of those work on wizards anymore. Ascendants are reliant on their thralls to protect them from wizards, as their powers aren't stronger than the Wizard's madness/reality bending powers/plasma addiction. The essentially pose the same threat to the wizard as a revhead Ian would, the real danger is the revs.

**Comment by Iamgoofball** [comment]
please ensure bullets fired by wizards will kill shadowlings

**Comment by Archie700** [comment]
Waitaminute why is syndicate lawyer in this pr

**Comment by Zilenan91** [comment]
It's not?

**Comment by kilkun** [comment]
my commit history is fucked up, ignore it.

the real question is why is my var-editing friendly dehydrated carp in this pr because im 99% certain that was already merged.

**Comment by AdamElTablawy** [comment]
For the record, what we on yog did was add orderable floodlights from cargo-they have a massive light radius and are unveilable, and have to be assembled when they come out of the crate so it's not an instant counter.

We also removed the jaunt ability for shadowlings-they instead have incorporeal movement jn dark tiles and dark tiles only, but are not projectile immune and can pull whatever they're dragging with them.

What this did was, for one, signficantly limit shadowling mobility-they actually did need thralls to disable lights for them to move around and get people not in maint, but gives them almost unrestricted mobility jn their territory.

Being able to pull things also allowed them to basically phase out of a wall, and drag an abductee through the wall into the darkness. It was pretty thematic at times, actually. Floodlights ended up as area denial for shadowlings-they couldn't run around through walls and over tables jn areas lit up, so to be able to go into areas cordoned off by floodlights, they needed thralls to help them-otherwise they risked being killed if they went into the light because they couldn't just glare and jaunt away. 

We also allowed icy veins to have a medium chance to extinguish flares and a low chance to extinguish torches.

All in all, those changes eliminated the stupid winrate for shadowling on yog, and it became more thrall dependent and 50/50 win/lose. You might want to go a similar route, because /tg/ shadowlings are reskinned wizards with thirty second stunmutes and effectively unkillable.

**Comment by kilkun** [comment]
that's a bit beyond the scope of this pr

let's see how it plays out before we do any major overhauls.

File: code/game/gamemodes/wizard/artefact.dm [hunk 8]
```diff
@@ -82,6 +82,55 @@
...
+	rend_desc = "You hear the sound of laughing slaughter demons."
+	spawn_fast = 1
+	var/captured_ascendants = 0
+
+/obj/item/weapon/veilrender/uncharged/attack_self(mob/user)
+	if(charges > 0 && captured_ascendants==2)
+		new /obj/effect/rend(get_turf(user), spawn_type, spawn_amt, rend_desc, spawn_fast)
+		charges--
+		captured_ascendants = 0
+		user << "<span class='boldannounce'>You merge the two ascendants' souls and drive them into bluespace. You hear blood curdling screams emitting from the souls and the hole to bluespace opens wider. Uh oh.</span>"//Blood dimension = bluespace because EMERGENT LORE EXPANSION is sorely needed
```
Comment (Man-Shroom-) [inline code/game/gamemodes/wizard/artefact.dm]: MUH LORE

IN KILKUN'S MAD MAD MIND, BLUESPACE AND NAR'SIE DIMENSION AND HELL ARE ALL THE SAME

File: code/game/gamemodes/wizard/artefact.dm [hunk 9]
```diff
@@ -82,6 +82,55 @@
...
+	item_state = "render"
+	force = 5 //shitty wizard build quality
+	throwforce = 0
+	w_class = 3
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	charges = 1
+	spawn_type = /obj/singularity/narsie/wizard
+	spawn_amt = 1
+	activate_descriptor = "the blood geometer's prison"
+	rend_desc = "You hear the sound of laughing slaughter demons."
```
Comment (Man-Shroom-) [inline code/game/gamemodes/wizard/artefact.dm]: MUH LORE

NAR'SIE DOES NOT LIVE IN THE SAME DIMENSION AS SLAUGHTER DEMONS, ONE LIVES OUTSIDE SPACE AND THE OTHER JUST LIVES IN HELL

Comment (Xhuis) [inline code/game/gamemodes/wizard/artefact.dm]: yo dude the ratvar lore isn't canon yet calm your tits

**Comment by Robustin** [comment]
1) Don't overbuff jaunt, when a sling jaunts you should have at least a rough idea of where it could have gone. 8 seconds means they could pretty much disappear to anywhere in the station every time they jaunt and even if you hit them with a long stun they will still be able to immediately jaunt and GTFO. Then you start hitting wizard-tiers of unkillability where you have to bomb the damn thing or else you'll never kill it. Adding 1 second to the duration and maybe 5 seconds off the cooldown would be enough. 

2) Thralls were one of the biggest issues. Being able to do a stun+mute long enough to cuff someone was a BIG DEAL and meant that slings would really snowball once they had a few competent thralls. IMO the lesser glare should have no mute and stun for a decent length in the dark but only last 1-2 seconds if the victim is well-lit. This should make for some fun tactics where you have to "ambush" someone with darkness by hitting a lightswitch or busting a light right before you glare them rather than just walking into every room and going "LOL LESSER GLARE" and cuff+remove headset while they're stunmuted. 

3) Where is this endgame going? The slings have already won right? And they can hide indefinitely in another dimensions... so how is sending the hunter going to be anything other than a meme?

**Comment by Iamgoofball** [comment]
@Robustin if the wizard hunter hits all the ascended slings with his drilldo veil render i think it would cause a victory condition

File: code/game/gamemodes/wizard/artefact.dm (hunk 9 above)
Comment (kilkun) [inline code/game/gamemodes/wizard/artefact.dm]: Fight me nerd. Demons = xenodimensional space aliens.

**Comment by kilkun** [comment]
@robustin 
This introduces a third stage in the gamemode. The Slings have "won" against the crew, but now they must face an extremely powerful foe.

@Iamgoofball 
The wizard can only absorb 2 souls with his knife. The third and final assling must survive Nar-Sie and the wizard if they took disintegrate or have a staff of change/chaos/death

**Comment by Xhuis** [comment]
I think the ascendant hunter is completely pointless. Not only can the ascendant enthrall the wizard, but the chain lightning stuns for a huge duration and they can indefinitely phase shift. Making the hunters immune to the abilities would make fighting them extremely frustrating because there would be no way to counter them. :-1:

**Comment by PKPenguin321** [comment]
but xhuis that just makes it cooler if the wizard somehow succeeds
i can already see the "stories of awesome" post

**Comment by Zilenan91** [comment]
The ascendant can't enthrall the wizard as far as I know, that's prevented.

Aside from that the ascendant can 2 shot the wizard if it gets in close but from what I understand the wiz can 1 shot the ascendants.

**Comment by kilkun** [comment]
@xhuis, what zile said. The ascendant can't do that, the only way it can hurt the wizard is if it hits it with it's regular mob attack, but if it's in range to do that the wizard is in range to steal it's soul. It's up to the thralls to protect the ascendant. Your job as an ascendant when the wizard is on station and not dead is to run and hide and wait for your thralls to kill it.

**Comment by TehZombehz** [comment]
Zilenan approves of this. That alone is a sign this is a terrible idea and shadowling should stay gone forever.

**Comment by kilkun** [comment]
Not constructive.

**Comment by coiax** [comment]
Gotta love people commenting without actually reading any of the changes. 10/10

File: code/game/gamemodes/wizard/artefact.dm (hunk 9 above)
Comment (optimumtact) [inline code/game/gamemodes/wizard/artefact.dm]: ratvar lore will never be canon :laughing:

**Comment by InsaneHyena** [comment]
Why is every PR you make such a meme, kilkun? Stop messing with round ending events. Just stop. Nar'Sie and ascended shadowlings are not meant to be beaten.

**Comment by kilkun** [comment]
do you have any better criticism other than whinging about "muh unbeatable super antag..."

**Comment by Xhuis** [comment]
He shouldn't need one. Ascendants are meant to be unstoppable and have no way to die outside of, say, the supermatter. I can't see why ascendant hunters are necessary in any way.

**Comment by kilkun** [comment]
because it's bad and boring fucking gameplay, enough to have warranted the removal of the gamemode from rotation.

**Comment by InsaneHyena** [comment]
Just like that deathsquad PR, this one can be summed up as "Kilkun tries to push yet another retarded change, nobody actually likes it, he throws a tantrum". Nothing about this change is needed or wanted, so just stop.

**Comment by kilkun** [comment]
Get salty all you want, but when you fail to bring valuable constructive criticism and instead project your asshurt all over my PR, I'm going to bitch you out for being a cancerous force in the community. That's the kind of behavior that drove Exadv1 away from helping out with development back when he still lended a helping hand to the downstreams. No one wants you here, no one is going to be convinced by your substanceless posts. You are contributing absolutely nothing and it's borderline offensive that you think that anybody gives a single fuck about your trash opinions when you attach nothing of value to them. Stop commenting with this garbage, you are actually getting my hopes up that something worth my time is getting posted.

**Comment by coiax** [comment]
> otherwise your comments are about as useful as @Iamgoofball 's shitposting.

image

**Comment by Iamgoofball** [comment]
yeah pretty much that

bring actual criticism to the field or fuck off

**Comment by Iamgoofball** [comment]
also :+1: good shit

**Comment by Man-Shroom-** [comment]
Don't buff shadowling or CHANGE THE DEEP LORE

other than that it's fine, sling hunter is good

**Comment by blobbernaut** [comment]
I like the idea of something that fights ascendants. They're my least favorite part of the mode anyway.

**Comment by KorPhaeron** [comment]
>I like the idea of something that fights ascendants. They're my least favorite part of the mode anyway.

Yeah I'll merge this if you remove the shadowling buffs they already had a 90% winrate or some shit

**Comment by kilkun** [comment]
Removed the balances

**Pull request closed without merging**